latex_access/tests/test_preprocessor.py

```diff
@@ -0,0 +1,70 @@
+import unittest
+import tempfile
+import pickle
+try:
+    from latex_access import preprocessor
+except ModuleNotFoundError:
+    incompatible = True
+else:
+    incompatible = False
+
+
+@unittest.skipIf(incompatible, "This module is not compatible with version 3 of Python.")
+class TestPreprocessor(unittest.TestCase):
+    def setUp(self):
+        """Creates a preprocessor instance."""
+        self.preprocessor = preprocessor.preprocessor()
+
+    def test_table(self):
+        """Tests creation of table attribute."""
+        self.assertEqual(self.preprocessor.table, {})
+
+    def test_add(self):
+        """Test adding command to table."""
+        self.preprocessor.add('foo', 'bar')
+        self.assertEqual(self.preprocessor.table['foo'], ('bar'))
+
+    def test_add_from_string(self):
+        """Tests adding command with args."""
+        self.preprocessor.add_from_string('foo', 'args', '1#2#3')
+        self.assertEqual(self.preprocessor.table['foo'], ['args', '1', 2, '', 3, ''])
+
+    def test_write(self):
+        """Tests that preprocessor entries can be saved to a file."""
+        file = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
I'm not a fan of leaving temporary files around if it can be helped. Would it be possible to use `tempfile.NamedTemporaryFile` as a context manager and check file content  inside `with` block? If not please explain in comments why  this has to be written the way it is.